powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java

```diff
@@ -1167,8 +1192,14 @@ public void expireOperations() {
         logger.debug("Running scheduled task for expiring operations");
 
         final PageRequest pageRequest = PageRequest.of(0, powerAuthServiceConfiguration.getExpireOperationsLimit());
-        try (final Stream<OperationEntity> pendingOperations = operationRepository.findExpiredPendingOperations(currentTimestamp, pageRequest)) {
-            pendingOperations.forEach(op -> expireOperation(op, currentTimestamp));
+        try (final Stream<OperationEntity> pendingOperations = operationQueryService.findExpiredPendingOperations(currentTimestamp, pageRequest)) {
+            pendingOperations.forEach(op -> {
+                try {
+                    expireOperation(op, currentTimestamp);
+                } catch (GenericServiceException e) {
+                    logger.error("Operation expiration failed, operation ID: {}", op.getId());
```

Comment:
I do not like this pattern when the whole story is not described in a single log entry, but we may fine-tune it later on.